x-enabled parameter does not support boolean values

**Description**

Services can be enabled/disabled using a `x-enabled` property. The property works with string values (`"false"`) but not boolean values (`false`).

**Steps to reproduce the issue:**

1. Create a Docker App with a service that is disabled using `x-enabled: false`.

**Describe the results you received:**

A conversion error occurred.

```
WARN[0000] network host: network.external.name is deprecated in favor of network.name 
panic: interface conversion: interface {} is bool, not string

goroutine 1 [running]:
github.com/docker/app/render.processEnabled(0xc4208640c0)
	/go/src/github.com/docker/app/render/render.go:115 +0x31b
github.com/docker/app/render.render(0xc42032ff40, 0x1, 0x1, 0xc420026300, 0xc42032ff40, 0x1, 0x1)
	/go/src/github.com/docker/app/render/render.go:93 +0x17c
github.com/docker/app/render.Render(0xc42016c070, 0xc4203bc120, 0x0, 0xc4203bc120, 0x4)
	/go/src/github.com/docker/app/render/render.go:79 +0x542
main.renderCmd.func1(0xc420372a00, 0xc4203798a0, 0x1, 0x1, 0x0, 0x0)
	/go/src/github.com/docker/app/cmd/docker-app/render.go:41 +0x1b2
github.com/docker/app/vendor/github.com/spf13/cobra.(*Command).execute(0xc420372a00, 0xc420379880, 0x1, 0x1, 0xc420372a00, 0xc420379880)
	/go/src/github.com/docker/app/vendor/github.com/spf13/cobra/command.go:762 +0x468
github.com/docker/app/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc420160a00, 0xc4203796c0, 0x20222e0, 0xc4203796d0)
	/go/src/github.com/docker/app/vendor/github.com/spf13/cobra/command.go:852 +0x30a
github.com/docker/app/vendor/github.com/spf13/cobra.(*Command).Execute(0xc420160a00, 0xc420160a00, 0x23c38a0)
	/go/src/github.com/docker/app/vendor/github.com/spf13/cobra/command.go:800 +0x2b
main.main()
	/go/src/github.com/docker/app/cmd/docker-app/main.go:18 +0xba
```

**Describe the results you expected:**

The service should be disabled in the rendered Compose file.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client:
 Version:           18.06.0-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        0ffa825
 Built:             Wed Jul 18 19:05:26 2018
 OS/Arch:           darwin/amd64
 Experimental:      true

Server:
 Engine:
  Version:          18.06.0-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       0ffa825
  Built:            Wed Jul 18 19:13:46 2018
  OS/Arch:          linux/amd64
  Experimental:     true
```

**Output of `docker-app version`:**

```
Version:      v0.4.1
Git commit:   48c0769c
Built:        Wed Aug 22 12:01:46 2018
OS/Arch:      darwin/amd64
Experimental: off
Renderers:    none
```

**Output of `docker info`:**

```
Containers: 41
 Running: 6
 Paused: 0
 Stopped: 35
Images: 254
Server Version: 18.06.0-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: active
 NodeID: w97q3z0azzcv970g3e5hn9ski
 Is Manager: true
 ClusterID: o1cdcrn2etmq3kcz2aapse3n4
 Managers: 1
 Nodes: 1
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 10
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
  Force Rotate: 0
 Autolock Managers: false
 Root Rotation In Progress: false
 Node Address: 192.168.65.3
 Manager Addresses:
  192.168.65.3:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: d64c661f1d51c48782c9cec8fda7604785f93587
runc version: 69663f0bd4b60df09991c08812a60108003fa340
init version: fec3683
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.93-linuxkit-aufs
Operating System: Docker for Mac
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 7.786GiB
Name: linuxkit-025000000001
ID: PJTX:MLTV:L2TJ:ISDE:ELPG:7QKG:G7HL:D34G:5ERF:SUIX:63WT:6WOR
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 102
 Goroutines: 268
 System Time: 2018-08-24T21:33:11.1974788Z
 EventsListeners: 6
HTTP Proxy: gateway.docker.internal:3128
HTTPS Proxy: gateway.docker.internal:3129
Registry: https://index.docker.io/v1/
Labels:
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Docker for Mac 18.06.0-ce-mac69 on macOS 10.13.6.

--- a/render/render.go
+++ b/render/render.go
@@ -90,7 +90,9 @@ func render(configFiles []composetypes.ConfigFile, finalEnv map[string]string) (
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to load Compose file")
 	}
-	processEnabled(rendered)
+	if err := processEnabled(rendered); err != nil {
+		return nil, err
+	}
 	return rendered, nil
 }
 
@@ -108,27 +110,44 @@ func errorIfMissing(substitution string, mapping composetemplate.Mapping) (strin
 	return value, true, nil
 }
 
-func processEnabled(config *composetypes.Config) {
+func processEnabled(config *composetypes.Config) error {
 	services := []composetypes.ServiceConfig{}
 	for _, service := range config.Services {
 		if service.Extras != nil {
-			if xEnabled, ok := service.Extras["x-enabled"]; ok && !isEnabled(xEnabled.(string)) {
-				continue
+			if xEnabled, ok := service.Extras["x-enabled"]; ok {
+				enabled, err := isEnabled(xEnabled)
+				if err != nil {
+					return err
+				}
+				if !enabled {
+					continue
+				}
 			}
 		}
 		services = append(services, service)
 	}
 	config.Services = services
+	return nil
 }
 
-func isEnabled(e string) bool {
-	e = strings.ToLower(e)
-	switch {
-	case e == "", e == "0", e == "false":
-		return false
-	case strings.HasPrefix(e, "!"):
-		return !isEnabled(e[1:])
-	default:
-		return true
+func isEnabled(e interface{}) (bool, error) {
+	switch v := e.(type) {
+	case string:
+		v = strings.ToLower(v)
+		switch {
+		case v == "", v == "0", v == "false":
+			return false, nil
+		case strings.HasPrefix(v, "!"):
+			nv, err := isEnabled(v[1:])
+			if err != nil {
+				return false, err
+			}
+			return !nv, nil
+		default:
+			return false, errors.Errorf("%s is not a valid value for x-enabled", e)
+		}
+	case bool:
+		return v, nil
 	}
+	return false, errors.Errorf("invalid type (%T) for x-enabled", e)
 }

--- a/render/render_test.go
+++ b/render/render_test.go
@@ -53,3 +53,26 @@ func TestRender(t *testing.T) {
 	assert.Check(t, is.Equal(c.Services[0].Image, "busybox:latest"))
 	assert.Check(t, is.DeepEqual([]string(c.Services[0].Command), []string{"-text", "baz"}))
 }
+
+func TestRenderEnabledFalse(t *testing.T) {
+	for _, tc := range []interface{}{false, "false"} {
+		configs := []composetypes.ConfigFile{
+			{
+				Config: map[string]interface{}{
+					"version": "3.7",
+					"services": map[string]interface{}{
+						"foo": map[string]interface{}{
+							"image":     "busybox",
+							"command":   []interface{}{"-text", "foo"},
+							"x-enabled": tc,
+						},
+					},
+				},
+			},
+		}
+		c, err := render(configs, map[string]string{})
+		assert.NilError(t, err)
+		assert.Check(t, is.Len(c.Services, 0))
+	}
+
+}

